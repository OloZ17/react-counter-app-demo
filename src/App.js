import React, { useState } from 'react';
import NavBar from './components/navbar';
import Counters from './components/counters';

const App = () => {
  const initialState = [
    { id: 1, value: 0 },
    { id: 2, value: 0 },
    { id: 3, value: 0 },
    { id: 4, value: 0 },
  ];

  const [counters, setCounters] = useState(initialState);

  const handleIncrement = (counter) => {
    const newCounters = [...counters];
    const index = newCounters.indexOf(counter);
    newCounters[index].value++;
    setCounters(newCounters);
  };

  const handleDecrement = (counter) => {
    const newCounters = [...counters];
    const index = newCounters.indexOf(counter);
    newCounters[index].value--;
    setCounters(newCounters);
  };

  const handleReset = () => {
    const resetCounters = counters.map((c) => {
      c.value = 0;
      return c;
    });
    setCounters(resetCounters);
  };

  const handleDelete = (counterId) => {
    const countersDelete = counters.filter((c) => c.id !== counterId);
    setCounters(countersDelete);
  };

  const handleRestart = () => {
    setCounters(initialState);
  };

  return (
    <div className="main__wrap">
      <main className="container">
        <div className="card__box">
          <NavBar totalCounters={counters.filter((c) => c.value > 0).length} />
          <Counters
            counters={counters}
            onReset={handleReset}
            onIncrement={handleIncrement}
            onDecrement={handleDecrement}
            onDelete={handleDelete}
            onRestart={handleRestart}
          />
        </div>
      </main>
    </div>
  );
};

export default App;
