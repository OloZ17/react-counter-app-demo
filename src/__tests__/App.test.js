import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

describe('Testing App() Component', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
