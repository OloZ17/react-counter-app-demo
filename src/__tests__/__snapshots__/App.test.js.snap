// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App() Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="main__wrap"
  >
    <main
      class="container"
    >
      <div
        class="card__box"
      >
        <nav
          class="navbar navbar-light"
        >
          <div
            class="navbar-brand"
          >
            <i
              aria-hidden="true"
              class="fa fa-shopping-cart fa-lg m-2"
            />
            <span
              class="badge badge-pill bg-info m-2"
              style="width: 50px; font-size: 24px;"
            >
              0
            </span>
            Items
          </div>
        </nav>
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-4 text-center"
            >
              <button
                class="btn btn-success m-2"
                data-testid="reset-button"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-refresh"
                />
              </button>
              <button
                class="btn btn-primary m-2"
                data-testid="restart-button"
                disabled=""
              >
                <i
                  aria-hidden="true"
                  class="fa fa-recycle"
                />
              </button>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="row justify-content-center"
            >
              <div
                class="col-4 text-center"
              >
                <span
                  class="badge m-2 bg-warning"
                  style="font-size: 21px; padding: 9px; vertical-align: middle;"
                >
                  Zero
                </span>
                <button
                  class="btn btn-secondary"
                  data-testid="increment-button"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-plus-circle"
                  />
                </button>
                <button
                  class="btn btn-info m-2"
                  data-testid="decrement-button"
                  disabled=""
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-minus-circle"
                  />
                </button>
                <button
                  class="btn btn-danger"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-trash-o"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="row justify-content-center"
            >
              <div
                class="col-4 text-center"
              >
                <span
                  class="badge m-2 bg-warning"
                  style="font-size: 21px; padding: 9px; vertical-align: middle;"
                >
                  Zero
                </span>
                <button
                  class="btn btn-secondary"
                  data-testid="increment-button"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-plus-circle"
                  />
                </button>
                <button
                  class="btn btn-info m-2"
                  data-testid="decrement-button"
                  disabled=""
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-minus-circle"
                  />
                </button>
                <button
                  class="btn btn-danger"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-trash-o"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="row justify-content-center"
            >
              <div
                class="col-4 text-center"
              >
                <span
                  class="badge m-2 bg-warning"
                  style="font-size: 21px; padding: 9px; vertical-align: middle;"
                >
                  Zero
                </span>
                <button
                  class="btn btn-secondary"
                  data-testid="increment-button"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-plus-circle"
                  />
                </button>
                <button
                  class="btn btn-info m-2"
                  data-testid="decrement-button"
                  disabled=""
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-minus-circle"
                  />
                </button>
                <button
                  class="btn btn-danger"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-trash-o"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="row justify-content-center"
            >
              <div
                class="col-4 text-center"
              >
                <span
                  class="badge m-2 bg-warning"
                  style="font-size: 21px; padding: 9px; vertical-align: middle;"
                >
                  Zero
                </span>
                <button
                  class="btn btn-secondary"
                  data-testid="increment-button"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-plus-circle"
                  />
                </button>
                <button
                  class="btn btn-info m-2"
                  data-testid="decrement-button"
                  disabled=""
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-minus-circle"
                  />
                </button>
                <button
                  class="btn btn-danger"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-trash-o"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
